cl/lib/elasticsearch_utils.py

```diff
@@ -1073,22 +1076,25 @@ def build_es_base_query(
     cd: CleanData,
     child_highlighting: bool = True,
     api_version: Literal["v3", "v4"] | None = None,
-) -> tuple[Search, QueryString | None]:
+    alerts: bool = False,
+) -> tuple[Search, QueryString | None, QueryString | None]:
```

Comment:
I think we should use a dataclass instead of a tuple as the return type for this helper method. During my initial review of the PR, I consistently assumed the parent query was the second element in the tuple. I only realized the correct order (child query, parent query) after referring to the docstring 😅 

Also, the addition of a new element to the tuple has increased the number of `_` variables used to discard unwanted values.

Reply:
I agree adding the dataclass as the return type for this method is a good idea, as it would make the code clearer when used. Thanks.